docs/markdown/Java and Scala/kotlin.md

```diff
@@ -0,0 +1,223 @@
+    artifact="guava",
+    version="31.0.1-jre",
+    # See the callout below for more information on the `packages` argument.
+    packages=["com.google.common.**"],
+)
+```
+
+Pants requires use of a lockfile for third-party dependencies. After adding or editing `jvm_artifact` targets, you
+will need to update affected lockfiles by running `./pants generate-lockfiles`. The default lockfile is located at
+`3rdparty/jvm/default.lock`, but it can be relocated (as well as additional resolves declared) via the
+[`[jvm].resolves` option](doc:reference-jvm#section-resolves).
+
+> 📘 Thirdparty symbols and the `packages` argument
+>
+> To efficiently determine which symbols are provided by third-party code (i.e., without hitting the network in order
+> to compute dependencies in the common case), Pants relies on a static mapping of which artifacts provide which
+> symbols, and defaults to treating each `jvm_artifact` as providing symbols within its `group`.
+>
+> The `packages` argument allows you to override which symbols a `jvm_artifact` provides. See the
+> [`jvm_artifact` docs](doc:reference-jvm_artifact#codepackagescode) for more information.
+
+### `resource` targets
+
+To have your code [load files as "resources"](https://docs.oracle.com/javase/8/docs/technotes/guides/lang/resources.html):
+
+1. Add a `resource` or `resources` target with the relevant files in the `source` / `sources` field, respectively.
+2. Ensure that [an appropriate `source_root`](doc:source-roots) is detected for the `resources` target, in order to
+   trim the relevant prefix from the filename to align with the layout of your JVM packages.
+3. Add that target to the `dependencies` field of the relevant JVM target (usually the one that uses the JVM APIs to
+   load the resource).
+
+For example:
+
+```toml pants.toml
+[source]
+# In order for the resource to be loadable as `org/pantsbuild/example/lib/hello.txt`,
+# the `/src/jvm/ prefix needs to be stripped.
+root_patterns = ["/src/*"]
+```
+```python src/jvm/org/pantsbuild/example/lib/BUILD
+kotlin_sources(dependencies=[":hello"])
+
+resources(name="hello", sources=["hello.txt"])
+```
+```java src/jvm/org/pantsbuild/example/lib/Loader.java
+package org.pantsbuild.example.lib
+
+import com.google.common.io.Resources
+
+fun load() {
+  ... = Resources.getResource(Loader.class, "hello.txt")
+}
+```
+```text src/jvm/org/pantsbuild/example/lib/hello.txt
+Hello world!
+```
+
+# Tasks
+
+## Compile code
+
+To manually check that sources compile, use `./pants check`:
+
+```
+# Check a single file
+❯ ./pants check src/jvm/org/pantsbuild/example/lib/ExampleLib.kt
+
+# Check files located recursively under a directory
+❯ ./pants check src/jvm::
+
+# Check the whole repository
+❯ ./pants check ::
+```
+
+## Run tests
+
+To run tests, use `./pants test`:
+
+```
+# Run a single test file
+❯ ./pants test tests/jvm/org/pantsbuild/example/lib/ExampleLibTest.kt
+
+# Test all files in a directory
+❯ ./pants test tests/jvm::
```

Comment:
That would result in testing all subdirs too

```suggestion
❯ ./pants test tests/jvm
```

Reply:
I believe that is the intent. I'll update the comment.